src/tests/nativeaot/SmokeTests/HardwareIntrinsics/X64Avx512.csproj

```diff
@@ -0,0 +1,33 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <CLRTestKind>BuildAndRun</CLRTestKind>
+    <CLRTestPriority>0</CLRTestPriority>
+    <CLRTestTargetUnsupported Condition="'$(TargetArchitecture)' != 'x64'">true</CLRTestTargetUnsupported>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <DefineConstants>$(DefineConstants);AVX512_INTRINSICS;VECTORT256_INTRINSICS</DefineConstants>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <IlcArg Include="--instruction-set:avx512f,avx512f_vl,avx512bw,avx512bw_vl,avx512cd,avx512cd_vl,avx512dq,avx512dq_vl" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <CLRTestBashPreCommands><![CDATA[
+$(CLRTestBashPreCommands)
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        exit 0
+    fi
```

Comment:
Could you make this a condition in `CLRTestTargetUnsupported` above? If that one is false, we don't even build this test - no point building this on the expensive mac machines if we don't ever expect to see an x64 mac with this.

Reply:
I can, but I think that doesn't work well with the various merged test groups and that we don't build on the machine we're running on. We rather build in a shared CI job and download the tests to each machine, where the runner then does the checks on "should run or not".

-- Looks like there is some failure I need to work through as well though. We're getting constant false where we should be getting dynamic checks...

Reply:
Fixed the failure.

There was a bug in `Avx512Vbmi` which I extracted to its own PR and the smoke test was expecting Avx2/AvxVnni to be opportunistic when Avx was supported, but they can't quite yet.

Reply:
> I can, but I think that doesn't work well with the various merged test groups and that we don't build on the machine we're running on. We rather build in a shared CI job and download the tests to each machine, where the runner then does the checks on "should run or not".

I'm not going to block on this, but these tests will never be part of a merged test group because they require separate compiler switches. They also don't do any target-machine-specific lightup either - when we're compiling the tests we already know we're compiling for mac.

To be clear what I'm asking for is to delete this part and instead update the `CLRTestTargetUnsupported` on line 6 in this file to also check `or '$(TargetsOSX)' == 'true'`

Reply:
>  but these tests will never be part of a merged test group because they require separate compiler switches. They also don't do any target-machine-specific lightup either - when we're compiling the tests we already know we're compiling for mac.

Gotcha. I was expecting this to hit the same problems we hit with the HWIntrinsic tests, where filters like `'$(TargetsOSX)' == 'true'` actually caused the tests to not compile at all and where filters like `CLRTestTargetUnsupported` weren't being respected in the sub-test groups, but only in the wrappers themselves.

> I'm not going to block on this

I'll go ahead and fix it now, this has to wait for sign-off from David before I can merge anyways

Reply:
Should be resolved now